UKHO.D365CallbackDistributorStub.API/UKHO.D365CallbackDistributorStub.API/Controllers/DistributorController.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using UKHO.D365CallbackDistributorStub.API.Models.Request;
+using UKHO.D365CallbackDistributorStub.API.Services;
+
+namespace UKHO.D365CallbackDistributorStub.API.Controllers
+{
+    [Route("/webhook/notification")]
+    [ApiController]
+    [ExcludeFromCodeCoverage]
+    public class DistributorController : BaseController<DistributorController>
+    {
+        private readonly ILogger<DistributorController> _logger;
+        private readonly DistributionService _distributionService;
+        public DistributorController(ILogger<DistributorController> logger, DistributionService distributionService)
+        {
+            _logger = logger;
+            _distributionService = distributionService;
+        }
+
+        [HttpOptions]
+        public IActionResult Options()
+        {
+            _logger.LogInformation("Distributor option accessed.");
+            using (StreamReader? reader = new(Request.Body, Encoding.UTF8))
+            {
+                string? webhookRequestOrigin = HttpContext.Request.Headers["WebHook-Request-Origin"].FirstOrDefault();
+                HttpContext.Response.Headers.Add("WebHook-Allowed-Rate", "*");
+                HttpContext.Response.Headers.Add("WebHook-Allowed-Origin", webhookRequestOrigin);
+            }
```

Comment:
`StreamReader? reader` is never used so can be removed please:
```suggestion
            string? webhookRequestOrigin = HttpContext.Request.Headers["WebHook-Request-Origin"].FirstOrDefault();
            HttpContext.Response.Headers.Add("WebHook-Allowed-Rate", "*");
            HttpContext.Response.Headers.Add("WebHook-Allowed-Origin", webhookRequestOrigin);
```

Reply:
Changes done